c7n/tags.py

```diff
@@ -671,3 +660,154 @@ def process(self, resources):
+    """Removes the specified tags from the specified resources.
+    """
+
+    batch_size = 20
+
+    permissions = ('resourcegroupstaggingapi:UntagResources',)
+
+    def process_resource_set(self, resource_set, tag_keys):
+        client = utils.local_session(
+            self.manager.session_factory).client('resourcegroupstaggingapi')
+
+        arns = []
+        for r in resource_set:
+            _id = r[self.id_key]
+            if 'arn' in _id[:3]:
+                arns.append(_id)
+            else:
+                # catch edge-case e.g. ID is like /hostedzone/ZXB8XVP18AKH8
+                _id = _id.split("/")[-1]
+                arns.append(self.manager.generate_arn(_id))
+
+        self.manager.retry(
+            client.untag_resources,
+            ResourceARNList=arns,
+            TagKeys=tag_keys)
+
+
+class UniversalTagDelayedAction(TagDelayedAction):
+    """Tag resources for future action.
+
+    :example:
+
+        .. code-block :: yaml
+
+            policies:
+            - name: ec2-mark-stop
+              resource: ec2
+              filters:
+                - type: image-age
+                  op: ge
+                  days: 90
+              actions:
+                - type: mark-for-op
+                  tag: custodian_cleanup
+                  op: terminate
+                  days: 4
+    """
+
+    batch_size = 20
+    concurrency = 2
+    permissions = ('resourcegroupstaggingapi:TagResources',)
+
+    def process(self, resources):
+        self.id_key = self.manager.get_model().id
+
+        # Move this to policy? / no resources bypasses actions?
+        if not len(resources):
+            return
+
+        msg_tmpl = self.data.get('msg', self.default_template)
+
+        op = self.data.get('op', 'stop')
+        tag = self.data.get('tag', DEFAULT_TAG)
+        date = self.data.get('days', 4)
+
+        n = datetime.now(tz=tzutc())
+        action_date = n + timedelta(days=date)
+        msg = msg_tmpl.format(
+            op=op, action_date=action_date.strftime('%Y/%m/%d'))
+
+        self.log.info("Tagging %d resources for %s on %s" % (
+            len(resources), op, action_date.strftime('%Y/%m/%d')))
+
+        tags = {tag: msg}
+
+        batch_size = self.data.get('batch_size', self.batch_size)
+
+        _common_tag_processer(
+            self.executor_factory, batch_size, self.concurrency,
+            self.process_resource_set, self.id_key, resources, tags, self.log)
+
+    def process_resource_set(self, resource_set, tags):
+        client = utils.local_session(
+            self.manager.session_factory).client('resourcegroupstaggingapi')
+
+        arns = []
+        for r in resource_set:
+            _id = r[self.id_key]
+            if 'arn' in _id[:3]:
+                arns.append(_id)
+            else:
+                # catch edge-case e.g. ID is like /hostedzone/ZXB8XVP18AKH8
+                _id = _id.split("/")[-1]
+                arns.append(self.manager.generate_arn(_id))
```

Comment:
also unclear if generate_arn is universally implemented if we're going to depend on it, we probably need a unit test that validates presence on resource managers or a generic implementation.